UnityProject/Assets/Scripts/Objects/Kitchen/InteractableGriddle.cs

```diff
@@ -86,5 +86,10 @@ public void ServerPerformInteraction(ContextMenuApply interaction)
 		}
 
 		#endregion Interaction-ContextMenu
+
+		public void Dispose()
```

Comment:
on destroy might be better

Reply:
since that's the Unity way of saying hey this is being destroyed

Reply:
I think I could add something like this to all of these classes:
```csharp
void OnDestroy()
{
    this.Dispose();
}
```

But the Dispose() method is a requirement of having the IDisposable interface, so I can't remove that.

Reply:
Just pushed that change, does that seem better?